### PR TITLE
Hubs/Scopes Merge 19 - Add `pushIsolationScope` and fork methods

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :bulb: Motivation and Context

`pushIsolationScope` forks both current and isolation scope and makes them the current scopes.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/main/java/io/sentry/ScopesAdapter.java [hunk 1]
```diff
@@ -190,6 +195,32 @@ public void flush(long timeoutMillis) {
     return Sentry.getCurrentScopes().clone();
   }
 
+  @Override
+  public @NotNull IScopes forkedScopes(@NotNull String creator) {
+    return Sentry.forkedScopes(creator);
+  }
+
+  @Override
+  public @NotNull IScopes forkedCurrentScope(@NotNull String creator) {
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/ScopesAdapter.java:204]: `l`: I'm wondering if we should have method overloads here instead. E.g. `.forkedScope(ScopeType type)`, Same could be done for the getters.

File: sentry/src/main/java/io/sentry/ScopesAdapter.java [hunk 2]
```diff
@@ -190,6 +195,32 @@ public void flush(long timeoutMillis) {
...
+  }
+
+  @Override
+  public @NotNull ISentryLifecycleToken makeCurrent() {
+    // TODO this wouldn't do anything since it replaced the current with the same Scopes
+    return NoOpScopesStorage.NoOpScopesLifecycleToken.getInstance();
+  }
+
+  @Override
+  public @NotNull IScope getScope() {
```
Comment (markushi) [inline sentry/src/main/java/io/sentry/ScopesAdapter.java:215]: What's the reason for adding those getters?

**Review by markushi — APPROVED** [review]

File: sentry/src/main/java/io/sentry/Hub.java [hunk 3]
```diff
@@ -539,6 +539,11 @@ public void removeExtra(final @NotNull String key) {
     return NoOpScopesStorage.NoOpScopesLifecycleToken.getInstance();
   }
 
+  @Override
+  public @NotNull ISentryLifecycleToken pushIsolationScope() {
+    return NoOpScopesStorage.NoOpScopesLifecycleToken.getInstance();
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/Hub.java:544]: Shouldn't this also be `Sentry.pushIsolationScope()`? Same as in HubAdapter?

File: sentry/src/main/java/io/sentry/Scopes.java [hunk 4]
```diff
@@ -602,12 +593,28 @@ public ISentryLifecycleToken pushScope() {
           .log(SentryLevel.WARNING, "Instance is disabled and this 'pushScope' call is a no-op.");
       return NoOpScopesStorage.NoOpScopesLifecycleToken.getInstance();
     } else {
-      Scopes scopes = this.forkedCurrentScope("pushScope");
+      final @NotNull IScopes scopes = this.forkedCurrentScope("pushScope");
```
Comment (lbloder) [inline sentry/src/main/java/io/sentry/Scopes.java:596]: Comment on method regarding the return of a lifecycle token can be removed

File: sentry/src/main/java/io/sentry/Hub.java (hunk 3 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/Hub.java:544]: `Hub` should be removed before releasing the major IMO.

File: sentry/src/main/java/io/sentry/Scopes.java (hunk 4 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/Scopes.java:596]: Already removed in a follow up PR

**Pull request merged**